src/pubd/repository.rs

```diff
@@ -364,18 +343,45 @@ impl RepositoryContent {
         jail: &uri::Rsync,
         config: &RepositoryRetentionConfig,
     ) -> KrillResult<()> {
+        // withdraw objects if any
         if let Ok(objects) = self.objects_for_publisher(name) {
             let withdraws = objects.elements().iter().map(|e| e.as_withdraw()).collect();
             let delta = DeltaElements::new(vec![], vec![], withdraws);
 
             self.rrdp.publish(delta, jail, config)?;
-            self.stats.remove_publisher(name, self.rrdp.notification());
-
-            self.write_repository(config)
-        } else {
-            // nothing to remove
-            Ok(())
+            self.write_repository(config)?;
         }
+        // remove publisher is present
+        self.publishers.remove(name);
```

Comment:
This looks like a logic fix unrelated to the PR? Is that correct?

Reply:
Yeah, that is correct. It was leaving the publisher in the map for the RepositoryContent, with no actual published objects. It doesn't really hurt to do so, but I noticed when touching the code, so figured I would just also remove the map entry.